src/test/java/org/numenta/nupic/network/LayerTest.java

```diff
@@ -1770,5 +1786,110 @@ public void testStringToInferenceTransformer() {
                                                         // Received a record yet.
         assertEquals("[42]", (Arrays.toString((int[])ff.inference.getSDR())));
     }
-   
+
+    @Test
+    public void testMakeClassifiers() {
+        // Setup Parameters
+        Parameters p = Parameters.getAllDefaultParameters();
+        Map<String, Class<? extends Classifier>> inferredFieldsMap = new HashMap<>();
+        inferredFieldsMap.put("field1", CLAClassifier.class);
+        inferredFieldsMap.put("field2", SDRClassifier.class);
+        inferredFieldsMap.put("field3", null);
+        p.set(KEY.INFERRED_FIELDS, inferredFieldsMap);
+
+        // Create MultiEncoder and add the fields' encoders to it
+        MultiEncoder me = MultiEncoder.builder().name("").build();
+        me.addEncoder(
+                "field1",
+                RandomDistributedScalarEncoder.builder().resolution(1).build()
+        );
+        me.addEncoder(
+                "field2",
+                RandomDistributedScalarEncoder.builder().resolution(1).build()
+        );
+        me.addEncoder(
+                "field3",
+                RandomDistributedScalarEncoder.builder().resolution(1).build()
+        );
+
+        // Create a Layer with Parameters and MultiEncoder
+        Layer<Map<String, Object>> l = new Layer<>(
+                p,
+                me,
+                new SpatialPooler(),
+                new TemporalMemory(),
+                true,
+                null
+        );
+
+        // Make sure the makeClassifiers() method matches each
+        // field to the specified Classifier type
+        NamedTuple nt = l.makeClassifiers(l.getEncoder());
+        assertEquals(nt.get("field1").getClass(), CLAClassifier.class);
+        assertEquals(nt.get("field2").getClass(), SDRClassifier.class);
+        assertEquals(nt.get("field3"), null);
+    }
+
+    @Test
+    public void TestMakeClassifiersWithNoInferredFieldsKey() {
+        // Setup Parameters
+        Parameters p = Parameters.getAllDefaultParameters();
+
+        // Create MultiEncoder
+        MultiEncoder me = MultiEncoder.builder().name("").build();
+
+        // Create a Layer with Parameters and MultiEncoder
+        Layer<Map<String, Object>> l = new Layer<>(
+                p,
+                me,
+                new SpatialPooler(),
+                new TemporalMemory(),
+                true,
+                null
+        );
+
+        // Make sure the makeClassifiers() method throws exception due to
+        // absence of KEY.INFERRED_FIELDS in the Parameters object
+        try {
+            NamedTuple nt = l.makeClassifiers(l.getEncoder());
```

Comment:
Please change to: 

l.makeClassifiers(l.getEncoder());

remove assignment to local variable so that the "unused" warning is removed?